prov/efa/src/efa_domain.h

```diff
@@ -33,6 +33,7 @@ struct efa_domain {
 	size_t ibv_mr_reg_sz;
 
 	/* Only valid for RDM EP type */
+	bool			rdm_ep;		/* Set to true for RDM domain. False otherwise. */
```

Comment:
A new field seemed better than comparing `fabric_attr->name` for every `fi_mr_reg` call

Reply:
Yeah, that was I thinking of too. I don't want to compare fabric name every time since `fi_mr_reg` is on the hot path for rdm code path

Reply:
Not related to this PR, but I find `rdm_mode` can be safely removed as it is not used.